lib/containers/common.pm

```diff
@@ -130,34 +130,7 @@ sub allow_selected_insecure_registries {
     my $runtime = $args{runtime};
     die "You must define the runtime!" unless $runtime;
     my $registry = registry_url();
-
-    assert_script_run "echo $runtime ...";
-    if ($runtime =~ /docker/) {
-        # Allow our internal 'insecure' registry
-        assert_script_run(
-'echo "{ \"debug\": true, \"insecure-registries\" : [\"localhost:5000\", \"registry.suse.de\", \"' . $registry . '\"] }" > /etc/docker/daemon.json');
-        assert_script_run('cat /etc/docker/daemon.json');
-        systemctl('restart docker');
-    } elsif ($runtime =~ /podman/) {
-        assert_script_run "curl " . data_url('containers/registries.conf') . " -o /etc/containers/registries.conf";
-        assert_script_run "chmod 644 /etc/containers/registries.conf";
-        file_content_replace("/etc/containers/registries.conf", REGISTRY => $registry);
-    } else {
-        die "Unsupported runtime - " . $runtime;
-    }
-}
-
-sub clean_container_host {
-    my %args    = @_;
-    my $runtime = $args{runtime};
-    die "You must define the runtime!" unless $runtime;
-    if ($runtime =~ /buildah/) {
-        assert_script_run("$runtime rm --all",          timeout => 180);
-        assert_script_run("$runtime rmi --all --force", timeout => 300);
-    } else {
-        assert_script_run("$runtime ps -q | xargs -r $runtime stop", 180);
-        assert_script_run("$runtime system prune -a -f",             300);
-    }
+    $runtime->setup_registry();
```

Comment:
Since this function only contains this unique call, I would suggest to remove this function and do the `setup_registry` call directly from the tests.

Reply:
good point. i just tried to avoid rename known functions which might had confused ppl. i will remove allow_selected_insecure_registries